src/main/java/uk/gov/hmcts/cp/cdk/batch/tasklet/GenerateAnswersTasklet.java

```diff
@@ -46,234 +46,268 @@ public class GenerateAnswersTasklet implements Tasklet {
-            new SingleColumnRowMapper<>(Integer.class);
+        new SingleColumnRowMapper<>(Integer.class);
 
-    private static MapSqlParameterSource reservationParams(
-            final UUID caseId, final UUID queryId, final UUID docId
-    ) {
+    private static MapSqlParameterSource reservationParams(final UUID caseId, final UUID queryId, final UUID docId) {
         return new MapSqlParameterSource()
-                .addValue("case_id", caseId)
-                .addValue("query_id", queryId)
-                .addValue("doc_id", docId);
+            .addValue("case_id", caseId)
+            .addValue("query_id", queryId)
+            .addValue("doc_id", docId);
     }
 
     private static MapSqlParameterSource answerParamsRow(
-            final UUID caseId,
-            final UUID queryId,
-            final Integer version,
-            final String answer,
-            final String llmInput,
-            final UUID docId
+        final UUID caseId,
+        final UUID queryId,
+        final Integer version,
+        final String answer,
+        final String llmInput,
+        final UUID docId
     ) {
         return new MapSqlParameterSource()
-                .addValue("case_id", caseId)
-                .addValue("query_id", queryId)
-                .addValue("version", version)
-                .addValue("answer", answer)
-                .addValue("llm_input", llmInput)
-                .addValue("doc_id", docId);
+            .addValue("case_id", caseId)
+            .addValue("query_id", queryId)
+            .addValue("version", version)
+            .addValue("answer", answer)
+            .addValue("llm_input", llmInput)
+            .addValue("doc_id", docId);
     }
 
     private static List<MetadataFilter> buildMetadataFilters(final UUID docId) {
         return Collections.singletonList(
-                new MetadataFilter().key("document_id").value(docId.toString())
+            new MetadataFilter().key("document_id").value(docId.toString())
         );
     }
 
     @Override
-    @SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.OnlyOneReturn"})
+    @SuppressWarnings({ "PMD.CyclomaticComplexity", "PMD.OnlyOneReturn" })
     public RepeatStatus execute(final StepContribution contribution, final ChunkContext chunkContext) {
         final ExecutionContext stepCtx = contribution.getStepExecution().getExecutionContext();
         final JobExecution jobExecution = contribution.getStepExecution().getJobExecution();
         final ExecutionContext jobCtx = jobExecution != null ? jobExecution.getExecutionContext() : new ExecutionContext();
 
-        final String docIdStr = stepCtx.getString(CTX_DOC_ID_KEY, null);
-        final String verifiedKey = CTX_UPLOAD_VERIFIED_KEY + ":" + docIdStr;
+        // Read from context
+        final String caseIdStr = stepCtx.getString(CTX_CASE_ID_KEY, null);
+        final String docIdStrCtx = stepCtx.getString(CTX_DOC_ID_KEY, null);
```

Comment:
you could store and retrieve any serializable object, so, can put UUID s directly and retrieve and avoid additional conversions